hashicorp/dev-portal
# feat: refine footer, full-width always

## 🔗 Relevant links

- [Preview link][/waypoint/docs] 🔎
- [Asana task - update footer to align with design][task-one] 🎟️
- [Asana task - refine content-footer spacing][task-two] 🎟️
- [Asana task - Sidebar height/versioning visual bugs][task-three] 🎟️ 

## 🗒️ What

Moves the footer in `SidebarSidecarLayout` to a full-width setup, below the sidebar-content area, tweaks spacing in that layout, and slightly tweaks sidecar-active logic to compensate.

## 🤷 Why

- To ensure footer content layout does not break in sidebar-sidecar layout (without requiring re-designs)
- To adjust the spacing between sidebar-sidecar layout's content and footer areas (has been flagged as a high priority to match this to the `128px` in the design spec)

## 🛠️ How

- In `SidebarSidecarLayout`, moves the footer out of the content area, to a full-width row below the sidebar-content section
- The above required some refactoring to our `VersionSelector` positioning within the sidebar. This refactor also ensures the version selector does not overlap and eat into the scrollable sidebar content space (more noticeable when the version selector is open, eg test [upstream on /waypoint/docs][upstream:/waypoint/docs]
- Tweaks spacing after sidebar-sidecar content from `50vh + 106px` to revised value from design of `128px`
- Tweaks our `use-active-section` intersection area to catch more "low on the page" headings (to try to mitigate the sidecar-active-section imperfection attributable to the previous spacing tweak)

## 🧪 Testing

- [ ] Visit a sidebar-sidecar layout page, such as [/waypoint/docs][/waypoint/docs]
    - Ensure the full-width footer looks and functions as expected
    - Ensure the sidebar works as expected on both desktop and mobile
    - The sidecar should function as expected, with the caveat that if the last heading section of a page has a small amount of content below it, it may not be highlighted.
- [ ] Visit the [home page][preview], and ensure the footer still works as expected

## 💭 Anything else?

Intent here is not to implement a GA-ready footer and whitespace below content, but rather to implement something that will work for our beta launch without significant breakage.

There's definitely valid debate on whether the extra whitespace we had previously, with the sidecar always working, is less broken than removing that whitespace and compromising slightly on sidecar active-section-highlighting. I've tried to mitigate the sidecar breakage in this PR by tweaking our `use-active-section` threshold (from top 40% of screen area, to top 60% of screen area).

[task-one]: https://app.asana.com/0/1201987349274776/1202074017129359/f
[task-two]: https://app.asana.com/0/1202097197789424/1202192173675062/f
[task-three]: https://app.asana.com/0/1202114367927919/1202422373192582/f
[preview]: https://dev-portal-git-zsrefine-footer-hashicorp.vercel.app
[/waypoint/docs]: https://dev-portal-git-zsrefine-footer-hashicorp.vercel.app/waypoint/docs
[upstream:/waypoint/docs]: https://developer.hashicorp.com/waypoint/docs
